File: JavaScript/8js 3/symmetricPoint.js
/*
|--------------------------------------------------------------------------
| Points of Reflection
|--------------------------------------------------------------------------
|
*/

function symmetricPoint(p, q) {
   return [2 * q[0] - p[0], 2 * q[1] - p[1]]
}


function symmetricPoint(p, q) {
   return p.map((a, i) => q[i] * 2 - a)
}


function symmetricPoint([x, y], [a, b]) {
   return [2 * a - x, 2 * b - y]
}


console.log(symmetricPoint([0, 0], [1, 1]), [2, 2]);
console.log(symmetricPoint([2, 6], [-2, -6]), [-6, -18]);
console.log(symmetricPoint([10, -10], [-10, 10]), [-30, 30]);
console.log(symmetricPoint([1, -35], [-12, 1]), [-25, 37]);
console.log(symmetricPoint([1000, 15], [-7, -214]), [-1014, -443]);
console.log(symmetricPoint([0, 0], [0, 0]), [0, 0]);

File: Python/8py 2/symmetric_point.py
"""
|--------------------------------------------------------------------------
| Points of Reflection
|--------------------------------------------------------------------------
|
"""

def symmetric_point(p, q):
   return [2 * q[0] - p[0], 2 * q[1] - p[1]]


def symmetric_point(p, q):
   return [2 * b - a for a, b in zip(p, q)]


def symmetric_point(p, q):
   return [2 * q[i] - p[i] for i in range(2)]


print(symmetric_point([0,0], [1,1]), [2, 2])
print(symmetric_point([2, 6], [-2, -6]), [-6, -18])
print(symmetric_point([10, -10], [-10, 10]), [-30, 30])
print(symmetric_point([1, -35], [-12, 1]), [-25, 37])
print(symmetric_point([1000, 15], [-7, -214]), [-1014, -443])
print(symmetric_point([0, 0], [0, 0]), [0, 0])